vllm/utils.py

```diff
@@ -172,16 +173,33 @@ def _async_wrapper(*args, **kwargs) -> asyncio.Future:
 
 
 def get_ip() -> str:
+    host_ip = os.environ.get("HOST_IP")
+    if host_ip:
+        return host_ip
+
+    # IP is not set, try to get it from the network interface
+
     # try ipv4
     s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
     try:
         s.connect(("8.8.8.8", 80))  # Doesn't need to be reachable
         return s.getsockname()[0]
-    except OSError:
-        # try ipv6
+    except Exception:
+        pass
+
+    # try ipv6
+    try:
         s = socket.socket(socket.AF_INET6, socket.SOCK_DGRAM)
         s.connect(("dns.google", 80))
```

Comment:
```suggestion
        s.connect(("2001:4860:4860::8888", 80))
```

Reply:
Is it ok to absorb https://github.com/vllm-project/vllm/pull/3176 into this PR 👀

Reply:
Quite confused. That PR https://github.com/vllm-project/vllm/pull/3176 was merged last week. Why doesn't it appear in `main` code right now ...